File: benchmark/polynom.pl
#!/usr/bin/perl -w

use strict;

sub poly($)
{
  my $n = 500000;
  my $x = $_[0];

  my $mu = 10;
  my $pu = 0;

  my @pol;

  foreach (0 .. $n - 1) {
      foreach (0 .. 99) {
	  $pol[$_] = $mu = ($mu + 2) / 2;
      }

      my $s = 0;
      foreach (0 .. 99) {
	  $s = $x * $s + $pol[$_];
      }

      $pu += $s;
  }

  return $pu;
} 

my $res;
for (1..4) {
    $res = poly(0.2);
    print "$res\n";
} 

File: benchmark/polynom.py
#!/usr/bin/python

n = 500000
x = 0.2

def t(x):
    mu = 10.0
    pu = 0.0
    pol = [0] * 100
    r = range(0,100)

    for i in range(0,n):
        for j in r:
            pol[j] = mu = (mu + 2.0) / 2.0
        su = 0.0
        for j in r:
            su = x * su + pol[j]
        pu = pu + su
    return pu

for i in range(2):
  print t(x)

